allennlp/data/data_iterator.py

```diff
@@ -52,9 +55,8 @@ def _create_batches(self, dataset: Dataset, shuffle: bool) -> List[List[Instance
 
     @classmethod
     def from_params(cls, params: Params):
-        from allennlp.experiments.registry import Registry
         # TODO(Mark): The adaptive iterator will need a bit of work here,
         # to retrieve the scaling function etc.
 
-        iterator_type = params.pop_choice("type", Registry.list_data_iterators())
-        return Registry.get_data_iterator(iterator_type)(**params.as_dict())  # type: ignore
+        iterator_type = params.pop_choice("type", cls.list_available())
+        return cls.by_name(iterator_type)(**params.as_dict())  # type: ignore
```

Comment:
Oh, right, we hadn't yet established the `from_params()` pattern when we wrote this code.  Can you add in a TODO to switch this to initialize `from_params()`?